### ISSUE TITLE
Ensure ORT to handle the output format 3 of the ScanCode 32 series

### ISSUE THREAD

**Issue by sschuberth** [oss-review-toolkit/ort#6617]
> Ok never mind, the issue is caused by scancode v32.x.x having a different output format, which ORT does not support yet.

Ping @maxhbr who I believe was experiencing something similar with the [latest ScanCode pre-release](https://github.com/nexB/scancode-toolkit/releases/tag/v32.0.0rc2). I'll create a new issue.

_Originally posted by @sschuberth in https://github.com/oss-review-toolkit/ort/issues/6585#issuecomment-1455643624_

**Comment by maxhbr** [comment]
My experienced behavior:

> using a ~2 week old ORT and https://github.com/nexB/scancode-toolkit/releases/tag/v32.0.0rc2 on windows, the scan result contained copyrights, but the licenses list was always `[]`. 
> But there were no issues or problems reported and the whole pipeline just ran through silently swallowing all scanner license information.

**Comment by bennati** [comment]
The issue is caused by this line https://github.com/oss-review-toolkit/ort/blob/56400c1f467c531e4977056e3db7a43a40d0d9f5/scanner/src/main/kotlin/scanners/scancode/ScanCodeResultParser.kt#L181 which creates an empty sequence.
The reason is that the Scancode output format has changed from
```
{
      "path": "LICENSE",
      "type": "file",
      "licenses": [
        {
          "key": "here-proprietary",
          "score": 100.0,
          "name": "HERE Proprietary License",
          "short_name": "HERE Proprietary",
          "category": "Commercial",
 ```
 
 To:
```
{
      "path": "LICENSE",
      "type": "file",
      ...
      "detected_license_expression": "here-proprietary",
      "detected_license_expression_spdx": "LicenseRef-scancode-here-proprietary",
      "license_detections": [
        {
          "license_expression": "here-proprietary",
          "matches": [
            {
              "score": 100.0,
              "start_line": 4,
              "end_line": 10,
              ...
              "license_expression": "here-proprietary",
              "rule_identifier": "here-proprietary.LICENSE",
              "rule_url": "https://github.com/nexB/scancode-toolkit/tree/develop/src/licensedcode/data/licenses/here-proprietary.LICENSE"
            }
          ]
        }
      ],
```

**Comment by sschuberth** [comment]
Work for this has been started in the [scancode-format-3](https://github.com/oss-review-toolkit/ort/tree/scancode-format-3) branch.

**Comment by sschuberth** [comment]
The implementation is ready, but its merge is currently blocked by the discussion in https://github.com/oss-review-toolkit/ort/pull/7300.
